Google.PowerShell.IntegrationTests/BigQuery/BigQuery.GcbqDataset.Tests.ps1

```diff
@@ -0,0 +1,85 @@
+﻿. $PSScriptRoot\..\BigQuery\GcbqCmdlets.ps1
+$project, $zone, $oldActiveConfig, $configName = Set-GCloudConfig
+
+Describe "New-GcbqDataset" {
+
+	It "should take a name, description, and time to make a dataset" {
+		$data = New-GcbqDataset "test_data_id1" -Name "Testdata" -Description "Some interesting data!" -Timeout 86400000
+		$data | Should Not BeNullOrEmpty
+		$data.DatasetReference.DatasetId | Should Be "test_data_id1"
+		$data.FriendlyName | Should Be "Testdata"
+		$data.Description | Should Be "Some interesting data!"
+		$data.DefaultTableExpirationMs | Should Be 86400000
+	}
+
+	It "should accept a dataset object from pipeline" {
+		$data = New-Object -TypeName Google.Apis.Bigquery.v2.Data.Dataset
+		$data.DatasetReference = New-Object -TypeName Google.Apis.Bigquery.v2.Data.DatasetReference
+		$data.DatasetReference.DatasetId = "test_data_id2"
+		$data.DatasetReference.ProjectId = $project
+		$data.FriendlyName = "PipeTest"
+		$data.Description = "Some cool data coming hot off the pipeline!"
+		$data | New-GcbqDataset -outvariable newdata
+		$newdata | Should Not BeNullOrEmpty
+		$newdata.DatasetReference.DatasetId | Should Be "test_data_id2"
+		$newdata.FriendlyName | Should Be "PipeTest"
+	}
+
+	It "should accept a dataset object from a command line argument" {
+		$data = New-Object -TypeName Google.Apis.Bigquery.v2.Data.Dataset
+		$data.DatasetReference = New-Object -TypeName Google.Apis.Bigquery.v2.Data.DatasetReference
+		$data.DatasetReference.DatasetId = "test_data_id3"
+		$data.DatasetReference.ProjectId = $project
+		$data.FriendlyName = "PipeTest"
+		$data.Description = "Some cool data coming hot off the pipeline!"
+		$newdata = New-GcbqDataset -Dataset $data
+		$newdata | Should Not BeNullOrEmpty
+		$newdata.DatasetReference.DatasetId | Should Be "test_data_id3"
+		$newdata.FriendlyName | Should Be "PipeTest"
+	}
+
+	It "should not work with no arguments (meaning no datasetId)" {
+		$data = New-GcbqDataset | Should Throw
```

Comment:
Can we check for what is thrown here? For example, Should Throw "error message".